engine/src/main/java/org/hibernate/validator/internal/util/Version.java

```diff
@@ -30,10 +43,21 @@ public static void touch() {
 	 * @return the Java release as an integer (e.g. 8 for Java 8)
 	 */
 	public static int getJavaRelease() {
-		// Will return something like 1.8
-		String[] specificationVersion = System.getProperty( "java.specification.version" ).split( "\\." );
+		return JAVA_RELEASE;
+	}
+
+	public static int determineJavaRelease(String specificationVersion) {
+		if ( specificationVersion != null && !specificationVersion.trim().isEmpty() ) {
+			Matcher matcher = JAVA_VERSION_PATTERN.matcher( specificationVersion );  //match 1.<number> or <number>
+
+			if ( matcher.find() ) {
+				return Integer.valueOf( matcher.group( 1 ) );
+			}
+		}
 
-		return Integer.parseInt( specificationVersion[1] );
+		// Cannot determine Java version; Assuming 1.6, not enabling the 1.8-only features
+		LOG.unknownJvmVersion( specificationVersion );
+		return 6;
```

Comment:
interesting, defaulting to 6 if there version string is unexpected. I guess that is reasonable.